### PR TITLE
[issue-1285] Ordering of transactions to maximize mining profits - work in progress

### PR DESCRIPTION
**Pull request by JoshMelton**
## PR description
Helps maximize profits for miners using the besu client by ordering transactions by gasPrice then time

NOTES: I am not a java dev, I just happened to see the issue and thought I might be able to contribute or at the very least help someone else contribute.

I also wasn't able to get things running locally, is there any good documentation for setting up for development? 

## Fixed Issue(s)
https://github.com/hyperledger/besu/issues/1285

## Changelog
(I wasn't sure about this, MadelineMurray posted a good link about that, but Im not much of a technical writer) 
- [X] I thought about the changelog

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java [hunk 1]
```diff
@@ -73,6 +74,7 @@
       new TreeSet<>(
           comparing(TransactionInfo::isReceivedFromLocalSource)
               .thenComparing(TransactionInfo::getSequence)
+              .thenComparing(TransactionInfo::getGasPrice)
```
Comment (RatanRSur) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java:77]: We want to order by gas prices first and then by the order in which they came in.

**Comment by RatanRSur** [comment]
Also, an entry in the changelog would be great as well because it's a behavior change in mining.

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java (hunk 1 above)
Comment (JoshMelton) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java:77]: ah, I see, I actually am just now figuring out how to run the tests for besu / reading into the first contribution workflow =)

I should have something shortly!

Comment (JoshMelton) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransactions.java]: I think I got the new ordering right =) I managed to run the test locally and they worked! 
I am not sure why they don't build via git-actions though?

**Comment by JoshMelton** [comment]
I made a few changes =) 

> Also, an entry in the changelog would be great as well because it's a behavior change in mining.

I took a look at the changelog.md, but how should I stucture this / add a new update the changelog?

**Comment by RatanRSur** [comment]
For the changelog, I would put it under `Additions and Improvements` because it's not a bug.

Also all of your commits need to be signed off with [DCO](https://wiki.hyperledger.org/display/BESU/DCO)'s. This will involve you rebasing those commits and force-pushing to this branch at some point.

**Comment by MadelineMurray** [comment]
@JoshMelton @RatanRSur -> this change will need an update to https://besu.hyperledger.org/en/stable/Concepts/Transactions/Transaction-Pool/#dropping-transactions-when-the-transaction-pool-is-full won't it? If you could confirm, I'll raise a docs issue for the update.

**Comment by RatanRSur** [comment]
Hey @JoshMelton, do you need any further assistance on this?

**Comment by RatanRSur** [comment]
Hey @JoshMelton, are you still able to address comments on this? If you're busy with other things, no worries. Just let me know and I'll be happy to take it over.

**Pull request closed without merging**